Assets/Tests/EditModeTests/PartitionTests.cs

```diff
@@ -0,0 +1,286 @@
+            Assert.Greater(intersectionBitmap.Count, 0);
+
+            // Calculate expected ratio: intersection count / total count of partition2's communication zone
+            var expectedRatio = (float)intersectionBitmap.Count / partition2.CommunicationZone.Count;
+            Assert.AreEqual(expectedRatio, partition1.CommunicationRatio[key]);
+
+            // Clean up
+            bitmap4.Dispose();
+            bitmap5.Dispose();
+            intersectionBitmap.Dispose();
+        }
+
+        [Test]
+        public void CalculateIntersectionAndRatio_ReturnsZeroRatio_WhenNoIntersection()
+        {
+            // Arrange
+            var partition1 = new Partition(1, new List<Vertex> { _vertices[0] }, // Only use vertex at (2,2)
+                new Dictionary<Vector2Int, Bitmap> { { new Vector2Int(2, 2), _communicationZones[new Vector2Int(2, 2)] } });
+
+            // Create a second partition with no intersection
+            var vertex = new Vertex(3, 1, new Vector2Int(9, 9));
+            var vertices2 = new List<Vertex> { vertex };
+
+            var communicationZones2 = new Dictionary<Vector2Int, Bitmap>();
+            var bitmap4 = new Bitmap(0, 0, _bitmapWidth, _bitmapHeight);
+            bitmap4.Set(9, 9);
+            communicationZones2[new Vector2Int(9, 9)] = bitmap4;
+
+            var partition2 = new Partition(2, vertices2, communicationZones2);
+
+            // Act
+            partition1.CalculateIntersectionAndRatio(partition2);
+
+            // Assert
+            var key = (1, 2);
+
+            // Verify dictionaries contain the key
+            Assert.IsTrue(partition1.IntersectionZones.ContainsKey(key));
+            Assert.IsTrue(partition1.CommunicationRatio.ContainsKey(key));
+
+            // Verify ratio is zero (no intersection)
+            Assert.AreEqual(0f, partition1.CommunicationRatio[key]);
+
+            // Verify intersection bitmap is empty
+            Assert.AreEqual(0, partition1.IntersectionZones[key].Count);
+
+            // Clean up
+            bitmap4.Dispose();
+            partition1.IntersectionZones[key].Dispose();
+        }
+
+        [Test]
+        public void CalculateIntersectionAndRatio_SkipsCalculation_WhenAlreadyCalculated()
+        {
+            // Arrange
+            var partition1 = new Partition(1, _vertices, _communicationZones);
+            var partition2 = new Partition(2, _vertices, _communicationZones);
+
+            // Set up the intersection zones and communication ratio dictionaries manually
+            var intersectionBitmap = new Bitmap(0, 0, _bitmapWidth, _bitmapHeight);
+            intersectionBitmap.Set(5, 5); // Just a sample intersection
+
+            // Initialize the properties through reflection since they are private
+            var intersectionZonesField = typeof(Partition).GetProperty("IntersectionZones");
+            var communicationRatioField = typeof(Partition).GetProperty("CommunicationRatio");
```

Comment:
This code uses reflection. This is really bad. Remove this unit test.

Reply:
Why is that bad?

Reply:
Reflection is a tool that should be very seldom used. It should be thinked of as a hack. And see how fragile it is. We removed the setters and now this dynamic crap does not work. I don't understand why we cannot test this without using reflection